src/input/ButtonConfigWidget.cc

```diff
@@ -224,3 +226,17 @@ void ButtonConfigWidget::initComboBox(QComboBox *comboBox, const Settings::Setti
 
 	updateComboBox(comboBox, entry);
 }
+
+void ButtonConfigWidget::updateStates(){
+	if(!initizalied) return;
+
+	int cnt = InputDriverManager::instance()->getButtonCnt();
+	for (int i=0;i<16;i++) {
+		auto label = this->findChild<QLabel *>(QString("labelInputButton%1").arg(i));
+		QString Style = ButtonConfigWidget::EmptyString;
+		if( cnt <= i){
+			Style = ButtonConfigWidget::DisabledStyleString;
```

Comment:
It might look cleaner to do a ternary expression and a single assignment here rather than reassigning in the if statement.